### ISSUE TITLE
WireGuard: multiple issues with DNS search order

### ISSUE THREAD

**Issue by khimaros** [celzero/rethink-app#1639]
the default wireguard app for android supports setting DNS search list by appending `,<domain>` to the DNS server address. Rethink doesn't properly parse these (eg. when imported) resulting in DNS resolution failures.

but also, more importantly, Rethink doesn't offer an alternative way to set the search order for DNS resolution. is this something that could be added? either by supporting the `,<domain>` syntax or via another knob?

**Comment by ignoramous** [comment]
Search domains aren't supported, yet (not in WireGuard, not in the VPN tunnel #226).

> resulting in DNS resolution failures.

We've since fixed parsing these (even if we don't enforce search domains, yet): https://github.com/celzero/rethink-app/issues/1483#issuecomment-2146238015

**Comment by wrobelda** [comment]
> We've since fixed parsing these

Are you sure? Because I just run into this issue after importing my Wireguard config with search domain configured. I only realized what was happening when I saw in log files it was trying to hit my search domain URL on 53 port.

**Comment by ignoramous** [comment]
> Are you sure? Because I just run into this issue after importing my Wireguard config with search domain configured.

You're right. We don't. We'll pick this regex up from the official WireGuard repository and that should it in the upcoming version `v060`.

```kt
    private static final Pattern VALID_HOSTNAME = Pattern.compile("^(?=.{1,255}$)[0-9A-Za-z](?:(?:[0-9A-Za-z]|-){0,61}[0-9A-Za-z])?(?:\\.[0-9A-Za-z](?:(?:[0-9A-Za-z]|-){0,61}[0-9A-Za-z])?)*\\.?$");
```
https://github.com/Windscribe/Android-App/blob/2c585dd5979ba93655853d56d849ed8ad2c964cb/wgtunnel/src/main/java/com/wireguard/config/InetAddresses.java#L26
